games-strategy/freeciv/freeciv-3.1.0_beta2.ebuild

```
@@ -0,0 +1,193 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-4 )
+
+inherit desktop lua-single meson xdg
+
+DESCRIPTION="Multiplayer strategy game (Civilization Clone)"
+HOMEPAGE="http://www.freeciv.org/"
+
+if [[ ${PV} == 9999 ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/freeciv/freeciv/"
+else
+	MY_PV="R${PV//./_}"
+	SRC_URI="https://github.com/freeciv/freeciv/archive/refs/tags/${MY_PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~ppc64 ~x86"
+	MY_P="${PN}-${MY_PV}"
+	S="${WORKDIR}/${MY_P}"
+fi
+
+LICENSE="GPL-2+"
+SLOT="0"
+IUSE="dedicated gtk3 gtk4 json mapimg modpack mysql nls +qt5 qt6 readline rule-editor sdl +sound +system-lua web-server "
+
+# I'm pretty sure that you can't build both qt flavours at the same time
+REQUIRED_USE="
+	system-lua? ( ${LUA_REQUIRED_USE} )
+	!dedicated? ( || ( gtk3 gtk4 qt5 qt6 sdl ) )
+	qt5?  ( !qt6 )
+	qt6?  ( !qt5 )
+"
+
+RDEPEND="
+	app-arch/bzip2
+	app-arch/xz-utils
+	app-arch/zstd:=
+	dev-db/sqlite:3
+	dev-libs/icu:=
+	net-misc/curl
+	sys-devel/libtool
+	sys-libs/zlib
+	!dedicated? (
+		media-libs/libpng
+		gtk3? ( x11-libs/gtk+:3 )
+		gtk4? ( gui-libs/gtk:4 )
+		mapimg? ( media-gfx/imagemagick:= )
+		nls? ( virtual/libintl )
+		qt5? (
+			dev-qt/qtcore:5
+			dev-qt/qtgui:5
+			dev-qt/qtwidgets:5
+		)
+		qt6? (
+			dev-qt/qtbase:6[gui,widgets]
+		)
+		sdl? (
+			media-libs/libsdl2[video]
+			media-libs/sdl2-gfx
+			media-libs/sdl2-image[png]
+			media-libs/sdl2-ttf
+		)
+		sound? (
+			media-libs/libsdl2[sound]
+			media-libs/sdl2-mixer[vorbis]
+		)
+	)
+	json? ( dev-libs/jansson:= )
+	readline? ( sys-libs/readline:= )
+	system-lua? (
+		${LUA_DEPS}
+	)
+"
+DEPEND="${RDEPEND}
+	!dedicated? ( x11-base/xorg-proto )
+"
+# Calls gzip during build
+BDEPEND="
+	app-arch/gzip
+	virtual/pkgconfig
+	nls? ( sys-devel/gettext )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-lua-search.patch
+)
+
+pkg_setup() {
+	use system-lua && lua-single_pkg_setup
+}
+
+src_configure() {
+	local myclient=() emesonargs=() myfcmp=()
+
+	# Upstream considers meson "experimental" until 3.2.0 according to their roadmap
+	emesonargs+=( -Dack_experimental=true )
+
+	# meson build drops the ability to _not_ build a server in favour
+	# of optionally replacing the server binary the freeciv-web backend
+	emesonargs+=(
+		$(meson_use web-server freeciv-web)
+	)
+
+	# Select any client backends that we want enabled; dedicated server shouldn't build a UI
+	# for sanity we'll build the modpack bin with the same UIs as the client.
+	# 'fcmp' = freeciv modpack (client) - gtk3, qt, cli, gtk4
+	if ! use dedicated ; then
+		# there's no SDL modpack backend; rather than incidentally pull in GTK3 (as is default)
+		# let's explicitly set the backend to CLI
+		use sdl && myclient+=( sdl2 ) && use modpack && myfcmp+=( cli )
```

What about a helper like this:
```suggestion
		freeciv_enable_flag() {
			local flag=$1
			local client_name=${2:-${flag}}
			local fcmp_name=${3:-${client_name}}
			
			if use ${flag} ; then
				myclient+=( ${client_name} )
				use modpack && myfcmp+=( ${fcmp_name} )
			fi
		}

		freeciv_enable_flag sdl sdl2 cli
		freeciv_enable_flag gtk3 gtk3.22 gtk3
		...
```